### PR TITLE
Commissions by partner

### PR DESCRIPTION
**Pull request by javierjcf**
We can set a diferent commision for the same agent in each partner.

### CODE REVIEW

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate this

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Why don't you use the same spelling in all places (`Agents & commissions`) and thus there's no similar strings to translate?

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate this

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate

File: sale_commission/i18n/es.po
Comment (pedrobaeza) [inline sale_commission/i18n/es.po]: Translate

File: sale_commission/models/res_partner.py
Comment (pedrobaeza) [inline sale_commission/models/res_partner.py]: This field should be in res.partner.agent level, not here. The idea is:
- You set the agents at customer (partner) level
- You check for each agent if they used the default commission set as agent level
- Or you specify a specific one for this customer

**Comment by javierjcf** [comment]
Ok,
I underestand if i do it this way, i must remove the agents field (because it make not sense have two diferent fields to set the agents in the partner) and put a post_init_hook to migrate the data from the agents fields.
Is that ok?

**Comment by pedrobaeza** [comment]
Yeah, that's it, @javierjcf!

**Comment by javierjcf** [comment]
Done.
I prefer a post-migration script instead of post_init_hook, because we only want do the script when the version change, and not after any installation.

File: sale_commission/__openerp__.py
Comment (pedrobaeza) [inline sale_commission/__openerp__.py]: It should be 8.0.3.0.0 as it includes a major change with migration script (https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers)

File: sale_commission/migrations/8.0.2.4.1/post-migration.py
Comment (pedrobaeza) [inline sale_commission/migrations/8.0.2.4.1/post-migration.py:13]: Instead null, put the partner company.

File: sale_commission/migrations/8.0.2.4.1/post-migration.py
Comment (pedrobaeza) [inline sale_commission/migrations/8.0.2.4.1/post-migration.py:14]: Isn't this table dropped when updating the module? If so, you have to rename it on the pre-migration script.

File: sale_commission/models/partner_agent.py
Comment (pedrobaeza) [inline sale_commission/models/partner_agent.py:2]: Don't add 2 copyright lines (and it's 2016 I suppose :wink: )

File: sale_commission/models/partner_agent.py
Comment (pedrobaeza) [inline sale_commission/models/partner_agent.py:1]: The file should be `res_partner_agent.py`

**Comment by pedrobaeza** [comment]
Thanks for the efforts you are making, Javier. Let's go for the last steps:
- You are making the commit messages in Spanish, but it should be in English. I suggest you squash the commit history and write a new global message in English.
- The coverage is dropping. Can you add tests for the non-covered parts for keeping it?

**Comment by javierjcf** [comment]
One line is difficult to cover (and i think  it was no covered before the pull request):
 if lines and lines[0]._name != 'account.invoice.line':
            return res
I think is not neccesarry this lines but i prefer don't remove it.

**Comment by javierjcf** [comment]
Hi Pedro, 
I check there is no need to pre_init hook, the table partner_agent_rel is not removed.
I check it updating the module from OCA 8.0 branch to my branch, in a clean database with demo data.
I see that the data was migrated and the many2many table already exists.

**Comment by DarkoNikolovski** [comment]
Hi All,

Is there any chance that this PR will be merged soon ?

**Comment by pedrobaeza** [comment]
There are conflicts, so you must rebase

**Comment by DarkoNikolovski** [comment]
Rebased locally. May I push the changes to 8.0 ?

**Comment by pedrobaeza** [comment]
You cannot push. It should be the author. Another option is to close this PR and you make another one with your rebase, superseeding this one.

**Comment by javierjcf** [comment]
Hi, 
I did rebase and push.

**Comment by DarkoNikolovski** [comment]
Thanks.

**Comment by DarkoNikolovski** [comment]
Any news when this PR would be merged ?

**Comment by pedrobaeza** [comment]
@DarkoNikolovski, you haven't even reviewed the PR, so please do it.

**Comment by DarkoNikolovski** [comment]
Sorry that I didn't comment here but I installed the module locally and tested it before two weeks. It was everything OK.

**Comment by yajo** [comment]
Hi, please make bots go :white_check_mark: and merge will come faster :wink:

**Comment by javierjcf** [comment]
Hi, i did tests for the uncover code of my PR.

**Comment by DarkoNikolovski** [comment]
Hi. I see all checks are passed . May we merge it now ?

File: sale_commission/security/ir.model.access.csv
Comment (atchuthan) [inline sale_commission/security/ir.model.access.csv:3]: Needs minimum READ access in base.group_user for agent_lines_company as it is added to res.partner using One2many field

**Review by atchuthan — APPROVED** [review]
Code review :+1: 
minor changes pointed out in security access and please test it using users other than Admin too.

**Comment by Derter** [comment]
Can we expect a merge here soon?

**Comment by pedrobaeza** [comment]
There are still comments unattended and 2 approvals are needed (one from one of the repo commiters).

**Comment by Derter** [comment]
Hello Guys,
did this PR failed now? Does it need new review? 

Whats the next step that needs to happen before we can merge?

**Comment by yajo** [comment]
Not sure why runbot's ❌... any clues?

**Comment by pedrobaeza** [comment]
```
2017-02-15 12:44:35,240 187 WARNING openerp_test openerp.models: method res.partner._check_employee: @constrains parameter 'employee' is not writeable
```

**Comment by Derter** [comment]
Would appreciate a merge of this one! Thank you.

**Comment by yajo** [comment]
I'm afraid we can't merge with a red runbot... You could try rebasing to see if it fixes it.

File: sale_commission/models/__init__.py
Comment (Daniel-CA) [inline sale_commission/models/__init__.py:11]: Add end line

File: sale_commission/views/sale_order_view.xml
Comment (scigghia) [inline sale_commission/views/sale_order_view.xml:16]: this line, give me an error opening a draft sale order.

File: sale_commission/views/sale_order_view.xml
Comment (scigghia) [inline sale_commission/views/sale_order_view.xml:24]: same as above

**Comment by pedrobaeza** [comment]
Are you still interested in this? CIs are red, there are conflicting files. Note that there's an alternative in #115

**Comment by pedrobaeza** [comment]
Closing as no answer.

**Pull request closed without merging**
